app/src/main/java/com/example/xpvehicles/fragments/RentingVehiclesFragment.java

```diff
@@ -76,6 +76,7 @@ public void done(List<RentVehicle> vehicles, ParseException e) {
                     Log.e(TAG, "Issue with getting the vehicles",e);
                     return;
                 }
+                adapter.clear();
                 if (vehicles.size() > 0) {
                     tvUserNoRentingVehicles.setVisibility(View.GONE);
                     adapter.addAll(vehicles);
```

Comment:
Why don't we just call `adapter.setVehicles(vehicles)`? let adapter handle the logic of whether it should add to vehicles or no op. Then you don't need to add this clear() in every call. You just need to call setVehicles outside of this if statement.